### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by amartya-rs**
### In this PR:
- added private routes
- added restricted routes
- authorization (login and signup)
### Preview: https://deploy-preview-9--dopeyard-ecom.netlify.app/

### CODE REVIEW

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Clean code. Good to merge 👍🏻

File: src/components/category-card/CategoryCard.jsx [hunk 1]
```diff
@@ -13,7 +13,6 @@ const CategoryCard = ({ card }) => {
          type: "FILTER_BY_CATEGORY",
          payload: card.categoryName.toLowerCase().replaceAll(" ", ""),
       });
-      console.log(card.categoryName);
```
Comment (Megha-Joshi) [inline src/components/category-card/CategoryCard.jsx:16]: Remove console.log

File: src/pages/login-page/LoginPage.jsx [hunk 2]
```diff
@@ -1,8 +1,67 @@
 import { TopNav } from "../../components";
-import { Link } from "react-router-dom";
```
Comment (Megha-Joshi) [inline src/pages/login-page/LoginPage.jsx:2]: You have imported Link 2 times. Just remove the one.

**Review by Megha-Joshi — APPROVED** [review]
Code looks clean. Good with comments. Can be merged!

**Review by archit7448 — APPROVED** [review]
- proper structure , authReducer manage very well 
- clean code 
- looks good to merge

File: src/components/category-card/CategoryCard.jsx (hunk 1 above)
Comment (amartya-rs) [reply src/components/category-card/CategoryCard.jsx:16]: Red line means the code has been removed.

File: src/pages/login-page/LoginPage.jsx (hunk 2 above)
Comment (amartya-rs) [reply src/pages/login-page/LoginPage.jsx:2]: Red line means the code has been removed

**Pull request merged**